proof-producer/bin/proof-producer/CMakeLists.txt

```diff
@@ -40,11 +40,14 @@ function(setup_proof_generator_target)
         src/main.cpp
     )
 
+    add_library(${ARG_TARGET_NAME}-lib INTERFACE)
+
     # Make sure to add these dependencies before the others, since for multi-threaded version dependency on
     # actor-zk must come first.
     if(ARG_ADDITIONAL_DEPENDENCIES)
         foreach(lib IN LISTS ARG_ADDITIONAL_DEPENDENCIES)
             target_link_libraries(${ARG_TARGET_NAME} ${lib})
+            target_link_libraries(${ARG_TARGET_NAME}-lib INTERFACE ${lib})
```

Comment:
Now I see a lot of duplications here. Let's reduce it and make the library serve not only external binaries but proof-producer itself:
- `proof-producer-lib` should be linked (`PRIVATE`) to `proof-producer`
- Public headers should be specified for the library only
- External dependencies should also be linked to the library. But I suppose that some of them (like `Boost::program_options`) are not required for the headers, so please check and link them separately to the binaries only.

Reply:
done